packages/plugins/typed-variable-modal/src/Modal.js

```diff
@@ -0,0 +1,372 @@
+   */
+  render() {
+    /*
+     * Creates the Modal. The generated modal looks like:
+     * <div class="blockly-modal-container" role="dialog">
+     *   <header class="blockly-moddal-header">
+     *     <h2 class="blockly-modal-header-title">Modal Name</h2>
+     *     <button class="blockly-modal-btn">X</button>
+     *   </header>
+     *   <div class="blockly-modal-content">
+     *   </div>
+     *   <div class="blockly-modal-footer">
+     *     <button class="blockly-modal-btn blockly-modal-btn-primary">OK</button>
+     *     <button class="blockly-modal-btn">Cancel</button>
+     *   </div>
+     * </div>
+     */
+
+    // Create Container
+    this.htmlDiv_ = document.createElement('div');
+    Blockly.utils.dom.addClass(this.htmlDiv_, 'blockly-modal-container');
+    this.htmlDiv_.setAttribute('role', 'dialog');
+    // End creating the container
+
+    // Create the header
+    const modalHeader = document.createElement('header');
+    Blockly.utils.dom.addClass(modalHeader, 'blockly-modal-header');
+
+    this.renderHeader_(modalHeader);
+
+    const exitButton = document.createElement('button');
+    Blockly.utils.dom.addClass(exitButton, 'blockly-modal-btn');
+    Blockly.utils.dom.addClass(exitButton, 'blockly-modal-btn-close');
+    this.addEvent_(exitButton, 'click', this, this.onCancel_);
+    modalHeader.appendChild(exitButton);
+    // End create header
+
+    // Create content
+    const modalContent = document.createElement('div');
+    Blockly.utils.dom.addClass(modalContent, 'blockly-modal-content');
+    this.renderContent_(modalContent);
+    // End creating content
+
+    // Create Footer
+    const modalFooter = document.createElement('footer');
+    Blockly.utils.dom.addClass(modalFooter, 'blockly-modal-footer');
+
+    this.renderFooter_(modalFooter);
+    // End creating footer
+
+    this.htmlDiv_.appendChild(modalHeader);
+    this.htmlDiv_.appendChild(modalContent);
+    this.htmlDiv_.appendChild(modalFooter);
+  }
+
+  /**
+   * Render content for the modal header.
+   * @param {HTMLElement} headerContainer The modal's header div.
+   * @protected
+   */
+  renderHeader_(headerContainer) {
+    const modalTitle = document.createElement('H2');
+    Blockly.utils.dom.addClass(modalTitle, 'blockly-modal-header-title');
+    modalTitle.appendChild(document.createTextNode(this.title_));
+    this.htmlDiv_.setAttribute('aria-labelledby', this.title_);
+    headerContainer.appendChild(modalTitle);
+  }
+
+  /**
+   * Render content for the modal content div.
+   * @param {HTMLDivElement} _contentContainer The modal's content div.
+   * @protected
+   */
+  renderContent_(_contentContainer) {
+    // No-op on the base class.
+  }
+
+  /**
+   * Render content for the modal footer.
+   * @param {HTMLElement} _footerContainer The modal's footer div.
+   * @protected
+   */
+  renderFooter_(_footerContainer) {
+    // No-op on the base class.
+  }
+}
+Blockly.Css.register([`
+    .blockly-modal-container {
```

Comment:
I'm seeing different styles of CSS here. In blockly we use camelcase, and not `-`. Any reason to not stick with the blockly style?

Reply:
We use the - in workspace search plugin. It seems like that is the most agreed upon style for css naming in the community and what the google style guide says as well. I am fine either way, but would prefer that we keep all the plugins the same. Depending on what we decide will update workspace search to match.

Reply:
Agreed on consistency, and I'm okay with diverging from blockly style guide (to google style guide) if we're consistent about it. Problem here is things like field_date and field_slider would have to change (only because they live in blockly-samples), and some css that lives in blockly will follow one pattern and others not. 

I would recommend we follow the blockly style guide for things that exist in Blockly. Obviously for ES6 style things, we would have to match the Google style guide.